### PR TITLE
chore: Drop dependency on pointy

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Drop dependency on pointy and use `conversion.Pointer` instead

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 1]
```diff
@@ -365,26 +364,26 @@ func TestAccClusterAdvancedCluster_advancedConf(t *testing.T) {
...
+			FailIndexKeyTooLong:              conversion.Pointer(false),
+			JavascriptEnabled:                conversion.Pointer(true),
 			MinimumEnabledTLSProtocol:        "TLS1_1",
-			NoTableScan:                      pointy.Bool(false),
-			OplogSizeMB:                      pointy.Int64(1000),
-			SampleRefreshIntervalBIConnector: pointy.Int64(310),
-			SampleSizeBIConnector:            pointy.Int64(110),
-			TransactionLifetimeLimitSeconds:  pointy.Int64(300),
+			NoTableScan:                      conversion.Pointer(false),
+			OplogSizeMB:                      conversion.Pointer[int64](1000),
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:371]: does it fail if [int64] is not specified?

File: internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go [hunk 2]
```diff
@@ -239,10 +238,10 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
...
 		AuthorizedUserFirstName: d.Get("authorized_user_first_name").(string),
 		AuthorizedUserLastName:  d.Get("authorized_user_last_name").(string),
-		CopyProtectionEnabled:   pointy.Bool(d.Get("copy_protection_enabled").(bool)),
-		EncryptionAtRestEnabled: pointy.Bool(d.Get("encryption_at_rest_enabled").(bool)),
-		PitEnabled:              pointy.Bool(d.Get("pit_enabled").(bool)),
-		RestoreWindowDays:       pointy.Int(cast.ToInt(d.Get("restore_window_days"))),
+		CopyProtectionEnabled:   conversion.Pointer(d.Get("copy_protection_enabled").(bool)),
+		EncryptionAtRestEnabled: conversion.Pointer(d.Get("encryption_at_rest_enabled").(bool)),
+		PitEnabled:              conversion.Pointer(d.Get("pit_enabled").(bool)),
+		RestoreWindowDays:       conversion.Pointer(cast.ToInt(d.Get("restore_window_days"))),
```
Comment (lantoli) [inline internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go:244]: not sure why it's not similar to other places, e.g.   d.Get("restore_window_days").(int)

File: internal/service/project/data_source_projects_test.go [hunk 3]
```diff
@@ -19,18 +19,18 @@ func TestAccProjectDSProjects_basic(t *testing.T) {
...
-		PreCheck:                 func() { acc.PreCheckBasic(t); acc.PreCheckProjectTeamsIdsWithMinCount(t, 2) },
+		PreCheck:                 func() { acc.PreCheckBasic(t); acc.PreCheckProjectTeamsIDsWithMinCount(t, 2) },
 		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccMongoDBAtlasProjectsConfigWithDS(projectName, orgID,
 					[]*admin.TeamRole{
 						{
-							TeamId:    conversion.StringPtr(acc.GetProjectTeamsIdsWithPos(0)),
+							TeamId:    conversion.StringPtr(acc.GetProjectTeamsIDsWithPos(0)),
```
Comment (lantoli) [inline internal/service/project/data_source_projects_test.go:29]: do you know why Ids didn't fail the linter before?

**Review by lantoli — APPROVED** [review]
thanks, this is great!

File: internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go (hunk 2 above)
Comment (gssbzn) [reply internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go:244]: 🤷  I'm just replacing the pointer, there seem sto be a combination of using cast and `.int` all over

File: internal/service/project/data_source_projects_test.go (hunk 3 above)
Comment (gssbzn) [reply internal/service/project/data_source_projects_test.go:29]: I have a newer linter locally which picks these now, it was a small drive by I didn't bother in trying to switch to your version of the linter

File: internal/service/advancedcluster/resource_advanced_cluster_test.go (hunk 1 above)
Comment (gssbzn) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go:371]: yes, OplogSizeMB is int64 and the literal `1000` is int so you need to provide the cast

**Pull request merged**